mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -29,12 +29,17 @@
 
     private Configuration configuration;
 
+    private boolean extensionNotLoaded;
+
     @Override
     public void afterProjectsRead(MavenSession session) throws MavenExecutionException {
 
         configuration = Configuration.load();
 
-        if (isSkipExtension()) {
+        extensionNotLoaded = isSkipExtension();
```

Comment:
I would hide it in `isSkipExtension` (also maybe called it `shouldSkipExtensionInstallation`)

```java
private Boolean skipExtensionInstallation;

private boolean shouldSkipExtensionInstallation() {
  if (skipExtensionInstallation == null) {
    // initialize
  }
  return skipExtensionInstallation;
}


```